source/common/tls/connection_info_impl_base.cc

```diff
@@ -77,6 +80,39 @@ const std::string& ConnectionInfoImplBase::sha256PeerCertificateDigest() const {
   return cached_sha_256_peer_certificate_digest_;
 }
 
+const std::string& ConnectionInfoImplBase::sha256PeerCertificateChainDigests() const {
+  if (!cached_sha_256_peer_certificate_digests_.empty()) {
+    return cached_sha_256_peer_certificate_digests_;
+  }
+
+  STACK_OF(X509)* cert_chain = SSL_get_peer_full_cert_chain(ssl());
+  if (cert_chain == nullptr) {
+    ASSERT(cached_sha_256_peer_certificate_digests_.empty());
+    return cached_sha_256_peer_certificate_digests_;
+  }
+
+  auto stack_len = sk_X509_num(cert_chain);
```

Comment:
This is largely the same code between the functions. Please create a helper to factor out the common code.

Reply:
Good idea - done